lib/views/frontend/spree/checkout/payment/_stripe.html.erb

```diff
@@ -1,71 +1,59 @@
 <%= render "spree/checkout/payment/gateway", payment_method: payment_method %>
 
-<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
-<script type="text/javascript">
-  Stripe.setPublishableKey("<%= payment_method.preferred_publishable_key %>");
-</script>
-
+<script type="text/javascript" src="https://js.stripe.com/v3/"></script>
 <script>
-  Spree.stripePaymentMethod = $('#payment_method_' + <%= payment_method.id %>);
+  $(() => {
+    Spree.stripePaymentMethod = $('#payment_method_' + <%= payment_method.id %>);
+    let stripe = Stripe("<%= payment_method.preferred_publishable_key %>");
+    let elements = stripe.elements();
+    let card = elements.create('card', {
+      style: {
+        base: {
+          lineHeight: "20px"
+        }
+      }
+    });
+
+    function stripeTokenHandler(token) {
+      // Insert the token ID into the form so it gets submitted to the server
+      let form = $(Spree.stripePaymentMethod.parents("form").get(0));
+      let paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
+      let addStripeHiddenInput = (name, value) => {
+        form.append($("<input type='hidden' name='payment_source[" + paymentMethodId + "][" + name + "]' value='" + value + "'/>"));
+      }
```

Comment:
FYI, you don't need the `{}` for single-line closures.  Unless you had a reason?

```suggestion
      const form = $(Spree.stripePaymentMethod.parents("form").get(0));
      const paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
      const addStripeHiddenInput = (name, value) => form.append($("<input type='hidden' name='payment_source[" + paymentMethodId + "][" + name + "]' value='" + value + "'/>"));
```

Reply:
I didn't like how long it was on the single line. That was the only reason